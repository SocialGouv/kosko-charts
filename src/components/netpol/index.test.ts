//

beforeEach(() => {
  jest.resetModules();
});

test("should create a network policy", async () => {
  const { create } = await import("./index");
  const netpol = create("some-app");
  expect(netpol).toMatchSnapshot();
});
