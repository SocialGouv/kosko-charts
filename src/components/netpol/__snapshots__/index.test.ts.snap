// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a network policy 1`] = `
Object {
  "apiVersion": "networking.k8s.io/v1",
  "kind": "NetworkPolicy",
  "metadata": Object {
    "name": "default",
    "namespace": "some-app",
  },
  "spec": Object {
    "ingress": Array [
      Object {
        "from": Array [
          Object {
            "namespaceSelector": Object {
              "matchLabels": Object {
                "network-policy/namespace": "some-app",
              },
            },
          },
        ],
      },
      Object {
        "from": Array [
          Object {
            "namespaceSelector": Object {
              "matchLabels": Object {
                "network-policy/source": "ingress-controller",
              },
            },
          },
        ],
      },
      Object {
        "from": Array [
          Object {
            "namespaceSelector": Object {
              "matchLabels": Object {
                "network-policy/source": "monitoring",
              },
            },
          },
        ],
      },
    ],
    "podSelector": Object {},
    "policyTypes": Array [
      "Ingress",
    ],
  },
}
`;
